src/Raven.Server/Documents/Patch/BlittableObjectInstance.cs

```diff
@@ -121,6 +124,171 @@ public BlittableObjectProperty(BlittableObjectInstance parent, string property)
                     }
                 }
             }
+            
+            private bool TryGetValueFromIndex(BlittableObjectInstance parent, string property, out JsValue value)
+            {
+                value = null;
+                if (parent.Projection?.MustExtractFromDocument == true)
+                    return false;
+
+                if (parent.IndexFields == null)
+                    return false;
+
+                if (parent.IndexFields.TryGetValue(_property, out var indexField) == false && parent.AnyDynamicIndexFields == false)
+                    return false;
+
+                bool isLucene = parent.IndexRetriever.LuceneDocument != null && parent.IndexRetriever.State != null;
```

Comment:
Isn't the first condition `parent.IndexRetriever.LuceneDocument != null` enough?

Reply:
Actually, in old code we had a bit different conditions:

https://github.com/ravendb/ravendb/pull/16088/files#diff-2e22ccd65749e9ac7c17344149bff4b1cb1c8daa40edebd3c4f2d7f83c67769dL148